bindings/python/convert.py

```diff
@@ -257,25 +248,25 @@ def convert_generic(model_id: str, folder: str, filenames: Set[str], token: Opti
                 sf_in_repo = f"{prefix}.safetensors"
             sf_filename = os.path.join(folder, sf_in_repo)
             try:
-                convert_file(pt_filename, sf_filename)
+                convert_file(pt_filename, sf_filename, discard_names=[])
                 operations.append(CommitOperationAdd(path_in_repo=sf_in_repo, path_or_fileobj=sf_filename))
             except Exception as e:
                 errors.append((pt_filename, e))
     return operations, errors
 
 
-def convert(api: "HfApi", model_id: str, force: bool = False) -> Tuple["CommitInfo", List[Tuple[str, "Exception"]]]:
+def convert(api: "HfApi", model_id: str, revision: Optional[str] = None, force: bool = False) -> Tuple["CommitInfo", List[Tuple[str, "Exception"]]]:
     pr_title = "Adding `safetensors` variant of this model"
-    info = api.model_info(model_id)
+    info = api.model_info(model_id, revision=revision)
     filenames = set(s.rfilename for s in info.siblings)
 
-    with TemporaryDirectory() as d:
+    with TemporaryDirectory(prefix=os.getenv("HF_HOME", "") + "/") as d:
```

Comment:
(nit) you can use `huggingface_hub.constants.HF_HOME` to retrieve the user hf home (will check for `HF_HOME` or `XDG_CACHE_HOME` env variable + default to `~/.cache/huggingface`)

(nit 2) I would set a default directory in `HF_HOME + "/safetensors_converter"` just in case the converter crash without cleaning the folder afterwards (at least all tmp directories will be in the same place)

Reply:
Indeed. I removed that part actually to keep everything in a real temporary folder (`/tmp`). It prevents cache reuse, but should make the OS clean up correctly.

I had set this up for testing to use a real SSD on the machine I was using (/tmp was mounted to a slower disk)